### PR TITLE
feat: [SRM-15134]: Azure metrics populating service instance field for CV

### PR DESCRIPTION
**Pull request by karansaraswat19**
## Checklist
- [ ] I've documented the changes in the PR description.
- [ ] I've tested this change either in PR or local environment.
- [ ] If hashcheck failed I followed [the checklist](https://harness.atlassian.net/wiki/spaces/DEL/pages/21016838831/PR+Codebasehash+Check+merge+checklist) and updated this PR description with my answers to make sure I'm not introducing any breaking changes.

## Comment Triggers
<details>
  <summary>Build triggers</summary>

- Feature build: `trigger feature-build`
- 
  Specific builds
  - `trigger manager`
  - `trigger dms`
  - `trigger ng_manager`
  - `trigger cvng `
  - `trigger cmdlib`
  - `trigger template_svc`
  - `trigger events_fmwrk_monitor`
  - `trigger event_server`
  - `trigger change_data_capture`
  -  Trigger multiple builds together: For eg: `trigger dms, manager`
- Immutable delegate `trigger publish-delegate`
</details>

## [Latest PR Check Triggers](https://github.com/harness/harness-core/blob/develop/.github/pull_request_template.md)

<details>
  <summary>PR Check triggers</summary>
You can run multiple PR check triggers by comma separating them in a single comment. e.g. `trigger ut0, ut1`

- Compile: `trigger compile`
- CodeformatCheckstyle: `trigger checkstylecodeformat`
  - CodeFormat: `trigger codeformat`
  - Checkstyle: `trigger checkstyle`
- MessageMetadata: `trigger messagecheck`
- File-Permission-Check: `trigger checkpermission`
- Recency: `trigger recency`
- BuildNumberMetadata: `trigger buildnum`
- Trigger CommonChecks: `trigger commonchecks`
- PMD: `trigger pmd`
- Copyright Check: `trigger copyrightcheck`
- Feature Name Check: `trigger featurenamecheck`
- UnitTests-ALL: `trigger uts`
- UnitTests-0: `trigger ut0`
- UnitTests-1: `trigger ut1`
- UnitTests-2: `trigger ut2`
- UnitTests-3: `trigger ut3`
- UnitTests-4: `trigger ut4`
- UnitTests-5: `trigger ut5`
- UnitTests-6: `trigger ut6`
- UnitTests-7: `trigger ut7`
- UnitTests-8: `trigger ut8`
- UnitTests-9: `trigger ut9`
- CodeBaseHash: `trigger codebasehash`
- CodeFormatCheckstyle: `trigger checkstylecodeformat`
- SonarScan: `trigger ss`
- GitLeaks: `trigger gitleaks`
- Trigger all Checks: `trigger smartchecks`
- Go Build: `trigger gobuild`
- Validate_Reviews: `trigger review`
- Module Dependency Check: `trigger mdc`
</details>

## PR check failures and solutions
https://harness.atlassian.net/wiki/spaces/BT/pages/21106884744/PR+Checks+-+Failures+and+Solutions

## [Contributor license agreement](https://github.com/harness/harness-core/blob/develop/CONTRIBUTOR_LICENSE_AGREEMENT.md)

- - -
This change is [Reviewable](https://reviewable.io/reviews/harness/harness-core/50767)

### CODE REVIEW

**Review by deepak-harness — CHANGES_REQUESTED** [review]
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/50767)* status: 22 file(s) to review. 1 change requests. 0 of 1 approvals obtained. 

___
*[`srm-service/modules/cv-nextgen-service/contracts/delegate/src/main/java/io/harness/cvng/beans/azure/AzureMetricsSampleDataRequest.java` line 30 at r2](https://reviewable.io/reviews/harness/harness-core/50767#-N_m7uOM6HSYGWg8VbQ-:-N_m7uOM6HSYGWg8VbQ0:bt7qrxg) ([raw file](https://github.com/harness/harness-core/blob/cacce9655e97598e721d44c1cbe858824620e608/srm-service/modules/cv-nextgen-service/contracts/delegate/src/main/java/io/harness/cvng/beans/azure/AzureMetricsSampleDataRequest.java#L30)):*
> ```Java
> @FieldNameConstants(innerTypeName = "AzureSampleDataRequestKeys")
> public class AzureMetricsSampleDataRequest extends AbstractAzureDataRequest {
>   String metricName;
> ```

aren't we adding aggregation to this
___
*[`srm-service/modules/cv-nextgen-service/service/src/main/resources/azure/metric-packs/default-custom-pack.yaml` line 1 at r2](https://reviewable.io/reviews/harness/harness-core/50767#-N_m8aMr4m-OAIlkyV38:-N_m8aMr4m-OAIlkyV39:bscl0xe) ([raw file](https://github.com/harness/harness-core/blob/cacce9655e97598e721d44c1cbe858824620e608/srm-service/modules/cv-nextgen-service/service/src/main/resources/azure/metric-packs/default-custom-pack.yaml#L1)):*
> ```YAML
> identifier: Custom
> ```

nit: what is the significance of this?

**Comment by karansaraswat19** [comment]
trigger uts_JDK11

**Comment by karansaraswat19** [comment]
trigger uts_JDK11

**Review by deepak-harness — APPROVED** [review]
> Reviewed 3 of 14 files at r2, 20 of 20 files at r5, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/50767)* status: :shipit: complete! ✓

**Comment by karansaraswat19** [comment]
trigger ut5

**Comment by karansaraswat19** [comment]
trigger ut6

**Comment by karansaraswat19** [comment]
trigger ut8

**Comment by karansaraswat19** [comment]
trigger ut5

**Comment by karansaraswat19** [comment]
trigger ut6

**Comment by karansaraswat19** [comment]
trigger ut8

**Comment by karansaraswat19** [comment]
trigger uts

**Comment by karansaraswat19** [comment]
trigger ut6_JDK11

File: srm-service/modules/cv-nextgen-service/service/src/main/java/io/harness/cvng/core/beans/healthsource/HealthSourceParamsDTO.java [hunk 1]
```diff
@@ -8,25 +8,39 @@
...
 
   public static HealthSourceParamsDTO getHealthSourceParamsDTO(HealthSourceParams healthSourceParams) {
+    if (healthSourceParams == null) {
+      healthSourceParams = HealthSourceParams.builder().build();
+    }
     return HealthSourceParamsDTO.builder()
-        .region(Optional.ofNullable(healthSourceParams).orElse(HealthSourceParams.builder().build()).getRegion())
+        .region(healthSourceParams.getRegion())
+        .metricName(healthSourceParams.getMetricName())
+        .metricNamespace(healthSourceParams.getMetricName())
```
Comment (n00bitax) [inline srm-service/modules/cv-nextgen-service/service/src/main/java/io/harness/cvng/core/beans/healthsource/HealthSourceParamsDTO.java:32]: why is name and namespace same, arent we going to change the metric name to something else

File: srm-service/modules/cv-nextgen-service/service/src/main/java/io/harness/cvng/core/beans/healthsource/HealthSourceParamsDTO.java [hunk 2]
```diff
@@ -8,25 +8,39 @@
...
+  AggregationType aggregationType;
 
   public static HealthSourceParamsDTO getHealthSourceParamsDTO(HealthSourceParams healthSourceParams) {
+    if (healthSourceParams == null) {
+      healthSourceParams = HealthSourceParams.builder().build();
+    }
     return HealthSourceParamsDTO.builder()
-        .region(Optional.ofNullable(healthSourceParams).orElse(HealthSourceParams.builder().build()).getRegion())
+        .region(healthSourceParams.getRegion())
+        .metricName(healthSourceParams.getMetricName())
```
Comment (n00bitax) [inline srm-service/modules/cv-nextgen-service/service/src/main/java/io/harness/cvng/core/beans/healthsource/HealthSourceParamsDTO.java:31]: looks suspicious that metric name is at health source level

**Pull request merged**